userspace/libscap/scap.c

```diff
@@ -432,6 +434,56 @@ scap_t* scap_open_live_int(char *error, int32_t *rc,
 				return NULL;
 			}
 
+			// Check the API version reported
+			api_version = ioctl(handle->m_devs[j].m_fd, PPM_IOCTL_GET_API_VERSION, 0);
+			if (api_version < 0)
+			{
+				snprintf(error, SCAP_LASTERR_SIZE, "Kernel module does not support PPM_IOCTL_GET_API_VERSION");
+				scap_close(handle);
+				*rc = SCAP_FAILURE;
+				return NULL;
+			}
+			if (handle->m_api_version != 0 && handle->m_api_version != api_version)
```

Comment:
I think this is an old version typo: i'd expect the code to perform a `memcpy` here, like we do in scap_bpf, instead of checking that m_api_version (empty here) is != from api_version.
Same below for schema_version.

@gnosek am i missing anything here?

Reply:
Let me take a look :)

Reply:
OK, there _is_ a bug but it's not what you described. The purpose of this check is to ensure all the devices support the same API version (with eBPF it's not an issue since there's just one eBPF device, not one per cpu).

The bug is we don't set `handle->m_api_version` right after this check so that the first device opened locks us into a particular API version.

All in all, it's an unlikely failure mode, so thanks for spotting it now before we hit the once-in-a-million situation :)

Reply:
> The purpose of this check is to ensure all the devices support the same API version

That makes a whole lot of sense indeed!
> The bug is we don't set handle->m_api_version right after this check so that the first device opened locks us into a particular API version.

Yep, the missing `memcpy` looked weird!
Care to add a comment too?

Reply:
I added a few words, does this help?

Reply:
Thank you very much! It is very helpful!